addons/survey/populate/survey_survey.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.tools import populate
+
+class Survey(models.Model):
+    _inherit = 'survey.survey'
+    _populate_sizes = {'small': 10, 'medium': 50, 'large': 100}
+
+    def _populate_factories(self):
+        random = populate.Random('survey_survey')
+        highest_code = int(self._read_group([], [], ['session_code:max'])[0][0])
+        user_ids = self.env['res.users'].search([]).ids
+        user_range = [_ for _ in range(0, 11)]
+        question_range = [_ for _ in range(5, 21)]
+        answer_range = [_ for _ in range(1, 5)]
+
+        def get_session_code(values=None, counter=None, **kwargs):
+            """Assign a unique number that is greater than the largest session
+            code already stored in DB"""
+            return highest_code + counter + 1
+
+        def get_question_and_page_ids(values=None, **kwargs):
+            """Create random number of questions and question's answers"""
+            return [
+                fields.Command.create({
+                    'title': f"Question {i + 1} for {values['title']}",
+                    'question_type': 'multiple_choice',
+                    'suggested_answer_ids': [fields.Command.create({
+                        'value': f"Answer {j + 1} of question {i + 1} for {values['title']} "
+                    }) for j in range(random.choice(answer_range))],
+                })
+                for i in range(random.choice(question_range))
+            ]
+
+        def get_user_ids(values=None, **kwargs):
+            """Restrict backend access of some surveys to a random number of
+                users
+            """
+            user_num = random.choice(user_range)
+            if user_num == 0:
+                return False
+
+            user_ids_set = set()
+            while len(user_ids_set) < user_num:
+                user_ids_set.add(random.choice(user_ids))
+            return list(user_ids_set)
```

Comment:
I think this is unnecessarily long. How about something similar to this ?
```suggestion
            nb_users = random.choice(n_survey_users_options)
            start_idx = random.randint(len(users) - nb_users); 
            return users[start_idx: start_idx + num_users].ids
```